addon/components/dashboard/materials.js

```diff
@@ -0,0 +1,166 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { isPresent } from '@ember/utils';
+import { restartableTask, timeout } from 'ember-concurrency';
+import { use } from 'ember-could-get-used-to-this';
+import AsyncProcess from 'ilios-common/classes/async-process';
+import moment from 'moment';
+
+const DEBOUNCE_DELAY = 250;
+
+export default class DashboardMaterialsComponent extends Component {
+  daysInAdvance = 60;
+  @service store;
+  @service fetch;
+  @service iliosConfig;
+  @tracked showAllMaterials = false;
+  @service currentUser;
+  @use _course = new AsyncProcess(() => [this.loadCourse.bind(this), this.args.courseIdFilter]);
+  @use _materials = new AsyncProcess(() => [
+    this.loadMaterials.bind(this),
+    this.args.showAllMaterials,
+  ]);
+
+  async loadMaterials() {
+    const user = await this.currentUser.getModel();
+    let url = `${this.iliosConfig.apiNameSpace}/usermaterials/${user.id}`;
+    if (!this.args.showAllMaterials) {
```

Comment:
I think maybe we should go ahead and load everything here and then filter it down based on `showAllMaterials`.

Reply:
impossible without changing what's being delivered on the frontend. the date range data points are not part of the materials payload.

Reply:
Ah, ok. Yes. I remember you were saying this. Ok. Something to fixup on the API. 🆒